packages/ExternalLink/src/ExternalLink.js

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import PropTypes from "prop-types";
+import tokens from "@paprika/tokens";
+import NewTabIcon from "@paprika/icon/lib/NewTab";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import * as styled from "./ExternalLink.styles";
+
+const propTypes = {
+  /** Text for aria-label. */
+  ariaText: PropTypes.string,
+
+  /** Link text for showing. */
+  children: PropTypes.string,
+
+  /** Should have an underline, by default it's true */
+  hasUnderline: PropTypes.bool,
+
+  /** Link url for the target. */
+  href: PropTypes.string.isRequired,
+};
+
+const defaultProps = {
+  ariaText: null,
+  children: null,
+  hasUnderline: true,
+};
+
+function ExternalLink(props) {
+  const externalLinkIconSize = 11;
+  const I18n = useI18n();
+  const { ariaText, children, hasUnderline, ...moreProps } = props;
+
+  const getContentText = () => {
+    return children || I18n.t("externalLink.view");
+  };
+
+  const handleSwallowClick = e => {
+    e.stopPropagation();
+  };
+
+  return (
+    <styled.ExternalLinkStyled
+      aria-label={ariaText || getContentText()}
+      onClick={handleSwallowClick}
+      rel="noopener noreferrer"
+      target="_blank"
+      externalLinkIconSize={externalLinkIconSize}
+      {...moreProps}
+    >
+      <styled.ExternalLinkContentStyled hasUnderline={hasUnderline}>
+        {getContentText()}
+      </styled.ExternalLinkContentStyled>
+      <NewTabIcon
+        css={styled.ExternalLinkIconStyles}
+        color={tokens.textColor.icon}
```

Comment:
I'm not sure what might be the best way to keep these "styles" in the `externalLink.style.js` as they really are props being applied to the component... I'm wondering if there is a way to apply props as styles in styled components, anyone any experience with that?

Reply:
Discussed with @mikrotron. The `color` prop can just be set as a `color` style instead on this component and it will be handled via the `currentColor` keyword, so the svg in the icon shall inherit the color from the style. This also means we don't need to apply styling props in this file, which is what i was trying to avoid. 👍